File: euler_003.js
const start = Date.now();

let n = 600851475143;
let upperBound = parseInt(Math.sqrt(n)) + 1;
let largestFactor = 1;
let sieve = Array(upperBound);
sieve.fill(true);

for (let i = 2;i < upperBound;i++) {
    if (sieve[i] && n % i === 0) {
        largestFactor = i;
        console.log(i);
        while (n % i === 0) {
            n = n / i;
        }
    }
    let j = 2;
    while (sieve[i] && j * i <= upperBound) {
        sieve[j * i] = false;
        j++;
    }
    if (n === 1) {
        break;
    }
}


const end = Date.now();

console.log("Execute time: ", end - start, " Largest factor is: ", largestFactor);

File: Euler/euler_003.java
import java.util.Arrays;

public class euler_003 {

    public static void main(String[] args) {

        long startTime = System.nanoTime();

        int upperBound = (int) Math.ceil(Math.sqrt((double) 600851475143L));

        int largestFactor = 1;

        boolean[] sieve = new boolean[upperBound + 1];
        Arrays.fill(sieve, true);

        for (int n = 3; n <= upperBound; n+=2) {
            if (sieve[n]) {
                int m = 2;
                while (m * n <= upperBound) {
                    sieve[m * n] = false;
                    m++;
                }
                if (600851475143L % n == 0) {
                    largestFactor = n;
                }
            }
        }

        long duration = System.nanoTime() - startTime;

        System.out.println("The largest prime factor is " + largestFactor + ". Runtime took around " + duration + " nanoseconds.");
    }
    
}
